build-tools/xaprepare/xaprepare/Application/Context.cs

```diff
@@ -765,16 +765,20 @@ public async Task<bool> Init (string scenarioName = null)
 
 			Tools.Init (this);
 
-			Banner ("Updating Git submodules");
```

Comment:
Why is it that `Context.Init()` -- which runs for *all* scenarios -- always updates git submodules?  Wouldn't it make sense to move this into the specific scenarios which requires it, instead of *all* scenarios, so that the `UpdateMono` scenario *doesn't* update git submodules?

This would avoid adding `KnownConditions.MonoUpdateKludge`.

Reply:
It makes sense to run submodule update only in the main scenario, but it wouldn't remove the need for `KnownConditions.MonoUpdateKludge` - the `BuildInfo` lines fail as well.

Reply:
Also, `ConfigureEnvironment` has to be avoided as well in this case, so the condition is still needed

Reply:
Can we make all of those steps use a scenario based condition rather than adding this new condition? It makes sense to skip all this stuff if we aren't running `Required` or `Standard` scenarios I think?